src/app/campaign/flight/flight.container.ts

```diff
@@ -94,11 +97,56 @@ export class FlightContainerComponent implements OnInit, OnDestroy {
   }
 
   flightUpdateFromForm({ flight, changed, valid }) {
+    this.loadAvailabilityAllocationIfChanged(flight);
     this.campaignStoreService.setFlight({ localFlight: flight, changed, valid }, this.currentFlightId);
-    this.campaignStoreService.loadAvailability(flight);
     this.currentInventoryUri$.next(flight.set_inventory_uri);
   }
 
+  loadAvailabilityAllocationIfChanged(flight: Flight) {
+    this.flightState$
+      .pipe(
+        filter((flightState: FlightState) => {
+          // determine if the availability fields are present and have changed since the last update from form
+          const { localFlight } = flightState;
+          // dates come back as Date but typed string on Flight
+          const flightStartAtDate = flight.startAt && new Date(flight.startAt.valueOf());
+          const flightEndAtDate = flight.endAt && new Date(flight.endAt.valueOf());
+          return (
+            flight.startAt &&
+            flight.endAt &&
+            flightStartAtDate.valueOf() !== flightEndAtDate.valueOf() &&
+            flight.set_inventory_uri &&
+            flight.zones &&
+            flight.zones.length > 0 &&
+            (flightStartAtDate.valueOf() !== new Date(localFlight.startAt).valueOf() ||
+              flightEndAtDate.valueOf() !== new Date(localFlight.endAt).valueOf() ||
+              flight.set_inventory_uri !== localFlight.set_inventory_uri ||
+              !flight.zones.every(zone => localFlight.zones.indexOf(zone) > -1))
```

Comment:
... and this is the actual condition we're checking for. The parens do help demarcate this, but I wonder if at this point (~10 LOC) it might be good for comprehension to pull this out into its own function and assign each of these conditional clauses to a variable with a descriptive name, or something like that.

Reply:
Or maybe just assigning two variables right here would do the trick, and pulling this out into its own function is overdoing it.